draft-ietf-jsonpath-base.md

```diff
@@ -614,111 +608,97 @@ Queries:
+{: title="Index wildcard selector examples"}
 
-### Dot Wildcard Selector {#wildcard}
+### Child Selector
 
 #### Syntax
-{: unnumbered}
+<!-- {: unnumbered} -->
 
-The dot wildcard selector has the form `.*` as defined in the
-following syntax:
+The child selector has the form `[<//PICKERS//>]` where `<//PICKERS//>` is a comma-delimited
+collection of one or more //PICKERS//.
+Each //PICKER// is defined below.
 
 ~~~~ abnf
-dot-wild-selector    = "." wildcard       ;  dot followed by asterisk
-wildcard             = "*"
+child-selector  = "[" S list-entry 1*(S "," S list-entry) S "]"
+
+list-entry     =  ( quoted-member-name /
+                    element-index      /
+                    slice-index /
+                    filter
+                  )
 ~~~~
 
 #### Semantics
-{: unnumbered}
+<!-- {: unnumbered} -->
 
-A `dot-wild-selector` acts as a wildcard by selecting the nodes of
-all member values of an object in its input nodelist as well as all
-element nodes of an array in its input nodelist.
+A child selector operates on objects and arrays only.
+It contains a comma-delimited collection of //PICKERS// to indicate which
+object members and array elements to selects.
 
-Applying the `dot-wild-selector` to a primitive JSON value (a number,
-a string, `true`, `false`, or `null`) selects no node.
+Each //PICKER// type functions differently and may be defined to operate on only
+objects, only arrays, or both.
 
-#### Examples
-{: unnumbered}
+//PICKERS// of different types may be combined within a single value selector.
 
-JSON:
-
-    {
-      "o": {"j": 1, "k": 2},
-      "a": [5, 3]
-    }
-
-Queries:
-
-| Query | Result | Result Paths | Comment |
-| :---: | ------ | :----------: | ------- |
-| `$.o.*` | `1` <br> `2` | `$['o']['j']` <br> `$['o']['k']` | Object values      |
-| `$.o.*` | `2` <br> `1` | `$['o']['k']` <br> `$['o']['j']` | Alternative result |
-| `$.a.*` | `5` <br> `3` | `$['a'][0]` <br> `$['a'][1]`     | Array members      |
-{: title="Dot wildcard selector examples"}
+The resulting nodelist of a child selector is the concatenation of
+the nodelists from each of its //PICKERS// in the order that the //PICKERS//
+appear in the list.
+Note that any node selected by more than one //PICKER// is kept
+as many times in the nodelist.
 
-### Index Selector
+#### Name //PICKER//
 
-#### Syntax {#syntax-index}
+##### Syntax {#syntax-name}
 {: unnumbered}
 
-An index selector `[<index>]` addresses at most one object member value or at most one array element value.
-
-~~~~ abnf
-index-selector      = "[" S (quoted-member-name / element-index) S "]"
-~~~~
+A name //PICKER// `'<name>'` addresses at most one object member value.
 
-Applying the `index-selector` to an object value in its input nodelist, a
-`quoted-member-name` string is required to select the corresponding
-member value.
+Applying the `quoted-member-name` to an object value in its input nodelist,
+it string is required to select the corresponding member value.
```

Comment:
//PICKER// descriptions need to avoid saying that a //PICKER// can _select_ a value, as this confuses //PICKERS// and selectors. I suggest using the term "match".